src/bindings/python/src/pyopenvino/graph/discrete_type_info.hpp

```diff
@@ -6,6 +6,26 @@
 
 #include <pybind11/pybind11.h>
 
+#include "openvino/core/type.hpp"
+#include <string>
 namespace py = pybind11;
 
+
+// DiscreteTypeInfo doesn't own provided memory. Wrapper allows to avoid leaks.
+class DiscreteTypeInfoWrapper : public ov::DiscreteTypeInfo {
+private:
+    const std::string name_str;
+    const std::string version_id_str;
+
+public:
+    DiscreteTypeInfoWrapper(std::string _name_str, std::string _version_id_str)
+        : DiscreteTypeInfo(nullptr, nullptr, nullptr),
+          name_str(std::move(_name_str)),
+          version_id_str(std::move(_version_id_str)) {
+        name = name_str.c_str();
+        version_id = version_id_str.c_str();
+    }
```

Comment:
Optional
```suggestion
    DiscreteTypeInfoWrapper(std::string _name_str, std::string _version_id_str)
        : DiscreteTypeInfo(),
          name_str(std::move(_name_str)),
          version_id_str(std::move(_version_id_str)) {
        name = name_str.c_str();
        version_id = version_id_str.c_str();
    }
```
Will default ctor  of `DiscreteTypeInfo` not set internal pointers to null? The 3rd parameter is optional and can be removed.

Reply:
I don't see that [here](https://github.com/akuporos/openvino/blob/akup/py-op-extension/src/core/include/openvino/core/type.hpp#L37).